### ISSUE TITLE
Support header and cookie mutators in Istio based JWT handler

### ISSUE THREAD

**Issue by strekm** [kyma-project/api-gateway#237]
**Description**

Based on [POC](https://github.com/kyma-project/api-gateway/issues/212) implement header and cookie mutators for Istio based JWT handler. Implementation should be the same as in ORY oathkeeper.  Other mutators: id_token and hydrator should in error message. Configuration should not change.

ACs:
- [x] mutator configuration validated 
- [x] only header and cookie are supported in this task
- [x] implementation for other types of handlers should not change
- [x] Istio JWT integration test enhanced

**Reasons**

Provide same functionality for mutators as ORY oathkeeper

**DoD:**
- [x] provide documentation
~~- [ ] publish release notes and _What's New_ updates for Kyma customers~~
- [x] provide unit tests
- [x] provide integration test
~~- [ ] verify [Get started guides](https://kyma-project.io/docs/kyma/latest/02-get-started/)~~
~~- [ ] verify resource limits~~
- [ ] [followup issue](https://github.com/kyma-project/api-gateway/issues/276)
~~- [ ] create release and bump in kyma~~

**Attachments**
https://www.ory.sh/docs/oathkeeper/pipeline/mutator
depends on: https://github.com/kyma-project/api-gateway/issues/212
part of: https://github.com/kyma-project/api-gateway/issues/30

**PR**
- https://github.com/kyma-project/api-gateway/pull/269

**Comment by triffer** [comment]
Validations that we need to discuss:
- Should it be allowed to set an empty header or cookie value, e.g. to reset it.
- Should validate that there is no duplicated header or cookie for a rule?
- Only one mutator handler of each type is allowed, so it's not possible to have a configuration like
```
  rules:
    - path: /cookies
      methods: ["GET"]
      mutators:
        - handler: cookie
          config:
            cookies:
              "a-cookie": "a"
        - handler: cookie
          config:
            cookies:
              "b-cookie": "b"
```

**Comment by triffer** [comment]
### Handling of existing headers and cookies
Ory overwrites headers and cookie. The default behaviour of Istio `Add` in [HeaderOperations](https://istio.io/latest/docs/reference/config/networking/virtual-service/#Headers-HeaderOperations) will add the header value comma-separated if it already exists.
This behaviour is an issue for handling the cookies with `Add`as it will result in 
```
"Cookie": "x-existing-cookie=value, x-added-by-mutator=value; x-another-one-added-by-mutator=value"
```
instead of
```
"Cookie": "x-existing-cookie=value; x-added-by-mutator=value; x-another-one-added-by-mutator=value"
```
